diff -Naur zato/code/zato-server/src/zato/server/connection/amqp_.py work/code/zato-server/src/zato/server/connection/amqp_.py
--- zato/code/zato-server/src/zato/server/connection/amqp_.py	2017-05-25 12:26:52.000000000 +0300
+++ work/code/zato-server/src/zato/server/connection/amqp_.py	2017-06-01 08:33:54.000000000 +0300
@@ -321,15 +321,20 @@
         _RECEIVED='RECEIVED', _ZATO_ACK_MODE_ACK=AMQP.ACK_MODE.ACK.id):
         """ Invoked each time a message is taken off an AMQP queue.
         """
-        self.on_message_callback(
-            channel_config['service_name'], body, channel=_CHANNEL_AMQP,
-            data_format=channel_config['data_format'],
-            zato_ctx={'zato.channel_item': {
-                'id': channel_config.id,
-                'name': channel_config.name,
-                'is_internal': False,
-                'amqp_msg': msg,
-            }})
+        try:
+            self.on_message_callback(
+                channel_config['service_name'], body, channel=_CHANNEL_AMQP,
+                data_format=channel_config['data_format'],
+                zato_ctx={'zato.channel_item': {
+                    'id': channel_config.id,
+                    'name': channel_config.name,
+                    'is_internal': False,
+                    'amqp_msg': msg,
+                }})
+        except Exception, e:
+            logger.warn('Exception in on_message_callback: %s', format_exc(e))
+            msg.reject()
+            return
 
         if msg._state == _RECEIVED:
             if channel_config['ack_mode'] == _ZATO_ACK_MODE_ACK:
