diff -Naur zato/code/zato-common/src/zato/common/odb/api.py work/code/zato-common/src/zato/common/odb/api.py
--- zato/code/zato-common/src/zato/common/odb/api.py	2017-05-25 09:26:52.701179450 +0000
+++ work/code/zato-common/src/zato/common/odb/api.py	2017-05-25 09:23:14.879167649 +0000
@@ -76,7 +76,7 @@
             self.session_initialized = True
 
     def session(self):
-        return self._Session()
+        return self._session
 
     def close(self):
         self._session.close()
